### PR TITLE
Add values files for users to install ambient when using helm

### PR DESCRIPTION
**Pull request by jacob-delgado**
**Please provide a description of this PR:**

Add helm values to simplify helm installation of ambient.

See https://github.com/istio/istio.io/pull/13871/ and https://github.com/istio/istio/blob/master/manifests/profiles/ambient.yaml

### CODE REVIEW

File: manifests/charts/istio-cni/ambient-values.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+cni:
+  # Configuration log level of istio-cni binary
+  # by default istio-cni send all logs to UDS server
+  # if want to see them you need change global.logging.level with cni:debug
+  logLevel: info
+
+  # Allow the istio-cni container to run in privileged mode, needed for some platforms (e.g. OpenShift)
+  privileged: false
```
Comment (howardjohn) [inline manifests/charts/istio-cni/ambient-values.yaml:8]: confusing; the comment suggests it is true but it is marked as false

File: manifests/charts/istio-control/istio-discovery/ambient-values.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    VERIFY_CERTIFICATE_AT_CLIENT: "true"
+    ENABLE_AUTO_SNI: "true"
+
+    PILOT_ENABLE_HBONE: "true"
+    CA_TRUSTED_NODE_ACCOUNTS: "istio-system/ztunnel,kube-system/ztunnel"
+    PILOT_ENABLE_AMBIENT_CONTROLLERS: "true"
+
+  # Sidecar injector mutating webhook configuration path value for the clientConfig.service field.
+  # Override to pass env variables, for example: /inject/cluster/remote/net/network2
+  injectionPath: "/inject"
```
Comment (howardjohn) [inline manifests/charts/istio-control/istio-discovery/ambient-values.yaml:16]: why is this here?

Comment (jacob-delgado) [inline manifests/charts/istio-control/istio-discovery/ambient-values.yaml:16]:
```suggestion
```

File: manifests/charts/istio-cni/ambient-values.yaml (hunk 1 above)
Comment (jacob-delgado) [inline manifests/charts/istio-cni/ambient-values.yaml:8]:
```suggestion
  privileged: true
```

File: manifests/charts/istio-control/istio-discovery/ambient-values.yaml (hunk 2 above)
Comment (jacob-delgado) [reply manifests/charts/istio-control/istio-discovery/ambient-values.yaml]: Bad copy/paste. Removed

File: manifests/charts/istio-control/istio-discovery/ambient-values.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#.Values.pilot for discovery and mesh wide config
```
Comment (hanxiaop) [inline manifests/charts/istio-control/istio-discovery/ambient-values.yaml:1]: I think one proxyconfig is missing in this file:
```
meshConfig:
  defaultConfig:
    proxyMetadata:
      ISTIO_META_ENABLE_HBONE: "true"
```

**Review by hanxiaop — APPROVED** [review]
LGTM, thanks!

**Comment by linsun** [comment]
Hi @jacob-delgado @hanxiaop maybe I'm missing something, i noticed enable_auto_sni is disabled by default in Istio. Any reason why this is enabled by default for ambient?

cc @kfaseela

**Comment by kfaseela** [comment]
@linsun auto_sni was supposed to be enabled by default by now, but need to fix some integration tests in Istio to get this running smoothly. Ambient does not have any dependency on these old tests, and since auto_sni and verify_cert_at_client adds better security, it was enabled by default. @howardjohn may be able to explain better

**Comment by linsun** [comment]
Thanks @kfaseela! I'm just worried about flipping this without sufficient testing.
